### ISSUE TITLE
Emoji picker in the chat

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/spreed#1130]
Specs are rather light.

We don't want to create a beast here, just help people with a dedicated Emoji input option.
Second step: solution how to display emojis when no emoji font is installed.

**Comment by mario** [comment]
See here for my suggestion to this problem: https://github.com/nextcloud/server/issues/10293

Then we could standardize the look of it across platforms and @jancborchardt seems to agree.

**Comment by skjnldsv** [comment]
Surprise surprise: https://github.com/DCzajkowski/vue-emoji-picker && https://github.com/jkchao/vue-emoji
https://codepen.io/DCzajkowski/pen/jzLzWp

**Comment by nickvergessen** [comment]
So blocked by #1347

**Comment by juliushaertl** [comment]
We also should ship the emoji assets with the server so we don't have the images duplicate in every app.

**Comment by dasunsrule32** [comment]
I didn't want to open another issue on this, but I still don't see a way to select emoji's in the web chat. I didn't see a configuration option to enable it either. Since this is closed, I'm assuming this has been resolved or am I missing something? Thanks :)

**Comment by nickvergessen** [comment]
It is not closed 🤔

**Comment by dasunsrule32** [comment]
@nickvergessen Yeah, I saw closed, then I saw the status was changed back to open. It's been a long day lol

**Comment by simonspa** [comment]
I'd be interested to work on this but I fear I would need some guidance. Where would this emoji picker go? A new Vue component in nextcloud-vue which can be re-used by spreed but also for comments etc?

**Comment by jancborchardt** [comment]
cc @ma12-co @skjnldsv @juliushaertl for guidance there. I would assume a Vue component indeed, because we do have an emoji picker in the Social app already cc @violoncelloCH

**Comment by ma12-co** [comment]
Hi @simonspa, thank you for considering this :) 
Let's see which libraries are available for vue and discuss which one suits our needs the best. @violoncelloCH  which one did you use?
The code is here: https://github.com/nextcloud/spreed/blob/master/src/components/NewMessageForm/NewMessageForm.vue
Also, please join our development chat https://cloud.nextcloud.com/call/c7fz9qpr if you need any help

**Comment by simonspa** [comment]
After some discussion in the chat it looks like https://www.npmjs.com/package/emoji-mart-vue-fast is the most promising candidate. I'll go ahead and play a bit with it and see where I get with it.

**Comment by skjnldsv** [comment]
License: BSD-3-Clause
Not sure we can, not familiar with this license!

**Comment by simonspa** [comment]
Certainly far from being an expert, but this is what GNU has to say about it:

>Modified BSD license (#ModifiedBSD)
> This is the original BSD license, modified by removal of the advertising clause. It is a lax, permissive non-copyleft free software license, compatible with the GNU GPL.

> GNU Affero General Public License (AGPL) version 3
> This is a free software, copyleft license. Its terms effectively consist of the terms of GPLv3

https://www.gnu.org/licenses/license-list.html

It would be more of a problem in the reverse case (NC being BSD, component being *GPL) I guess.

**Comment by mario** [comment]
BSD is one of the most permissive licences. We can definitely use it.

**Comment by nickvergessen** [comment]
Confirmed by @schiessle we can use it

**Comment by violoncelloCH** [comment]
at least the original implementation of the emoji picker in social was done by @juliushaertl ... there we use [vue-emoji-picker](https://www.npmjs.com/package/vue-emoji-picker) -- see here: https://github.com/nextcloud/social/blob/52398387c5f062a34dcdd379243c6589aa89f47f/src/components/Composer.vue#L55-L81
I guess it would make sense to implement this as a reusable vue-component, so we can use the same one in all places?

**Comment by nickvergessen** [comment]
> I guess it would make sense to implement this as a reusable vue-component, so we can use the same one in all places?

Yes, we had a quick chat about it this morning. It will be added in the `@nextcloud/vue` package

**Comment by nickvergessen** [comment]
@simonspa  any progress? :)

**Comment by simonspa** [comment]
Yes, some, I started writing the component but then didn't have enough time to finish it. I might have the next days though - and I might hang around in the community chat to ask noob questions about Vue. :)

**Comment by georgehrke** [comment]
@simonspa Could you give me quick status update on the emoji picker? I need one for https://github.com/nextcloud/server/pull/21186

If you need help or have any questions, please don't hesitate to ask. :)

**Comment by nickvergessen** [comment]
Done via #3994
